portal/src/util/projectname.ts

```diff
@@ -0,0 +1,28 @@
+// ref https://github.com/bitcoinjs/bip39/blob/master/src/wordlists/english.json
+import * as wordlist from "./wordlist.json";
+
+export function determineWord(index: number): string {
+  return wordlist[index];
+}
+
+export function getRandom32BitsNumber(): number {
+  const randomBuffer = new Uint32Array(1);
+  window.crypto.getRandomValues(randomBuffer);
+  return randomBuffer[0];
+}
+
+export function maskNumber(num: number, startAt: number, bits: number): number {
+  return (num >> startAt) & ((1 << bits) - 1);
+}
+
+export function randomProjectName(): string {
```

Comment:
Instead of testing this function, test this `export function deterministicProject(n: number): string`. Define `randomProjectName` as

```typescript
export function randomProjectName(): string {
  return deterministicProjectName(getRandom32BitsNumber());
}
```